source/frontend/src/features/leases/detail/LeasePages/details/DetailConsultation.tsx

```diff
@@ -0,0 +1,66 @@
+import * as API from 'constants/API';
+import { Section } from 'features/mapSideBar/tabs/Section';
+import { SectionField } from 'features/mapSideBar/tabs/SectionField';
+import { useFormikContext } from 'formik';
+import useLookupCodeHelpers from 'hooks/useLookupCodeHelpers';
+import { IFormLease } from 'interfaces';
+import { Api_LeaseConsultation } from 'models/api/Lease';
+import * as React from 'react';
+export interface IDetailConsultationProps {
+  nameSpace?: string;
+}
+
+/**
+ * Sub-form containing lease detail administration fields
+ * @param {IDetailConsultationProps} param0
+ */
+export const DetailConsultation: React.FunctionComponent<
+  React.PropsWithChildren<IDetailConsultationProps>
+> = ({ nameSpace }) => {
+  const { values, setFieldValue } = useFormikContext<IFormLease>();
+
+  const { getByType } = useLookupCodeHelpers();
+  const consultationTypes = getByType(API.CONSULTATION_TYPES);
+
+  // Not all consultations might be comming from the backend. Add the ones missing.
+  if (values.consultations.length !== consultationTypes.length) {
+    const newConsultations: Api_LeaseConsultation[] = [];
+
+    consultationTypes.forEach(x => {
+      const newConsultation: Api_LeaseConsultation = {
+        id: 0,
+        parentLeaseId: values.id || 0,
+        consultationType: { id: x.id.toString(), description: x.name },
+        consultationStatusType: { id: 'UNKNOWN', description: 'Unknown' },
+        rowVersion: 0,
+      };
+
+      // If there is a consultation with the type, set the status to the existing one
+      let existingConsultation = values.consultations.find(y => y.consultationType?.id === x.id);
+      if (existingConsultation !== undefined) {
```

Comment:
this seems pretty similar to the logic in the edit, would it be worth extracting this so that any fixes/enhancements can be made in one area instead of two?

Reply:
I agree, they are both almost the same. The reason that I didn't do that is because the lease code could use some refactoring and this could be part of it. If you feel strongly about it I can do that now.